grails-app/services/org/transmartproject/db/RestExportService.groovy

```diff
@@ -22,52 +23,23 @@ class RestExportService {
 
     ConceptsResource conceptsResourceService
 
+    int cohortNumberID
+
     List<File> export(arguments) {
         DataExportFetchTask task = dataExportFetchTaskFactory.createTask(arguments)
         task.getTsv()
     }
 
-    def formatDataTypes(List datatypes){
-        //Maybe make this a marshaller.
-        def returnDataTypeList = []
-        def dataTypesList = []
-        datatypes.each { dataTypeMap ->
-            dataTypeMap.each {key, value->
-                if (key in dataTypesList){
-                    returnDataTypeList.each { map ->
-                        if (key in map.values()){
-                            value.each{it.remove('datatypeCode')}
-                            map.get('cohorts').add([concepts:value])
-                        }
-                    }
-                } else{
-                    dataTypesList.add(key)
-                    def datatypeCode = value.collect({it ->
-                        it.get('datatypeCode')})
-                    value.each{it.remove('datatypeCode')}
-                    def datatypeMap = [dataType:key,
-                                       dataTypeCode: datatypeCode[0],
-                                       cohorts:[[concepts:value]]]
-                    returnDataTypeList.add(datatypeMap)
-                }
-            }
-        }
-        returnDataTypeList
-    }
-
-
-    def getDataTypes(List conceptKeysList){
-        List cohortDataTypes = []
-        Map datatypesMap = [:]
+    public def getDataTypes(List conceptKeysList, List dataTypes, Integer cohortNumber){
```

Comment:
some small remarks:
- use cmd+alt+L (in your IntelliJ on Mac at least) to format your code nicely. This way you will avoid style guide mistakes (we follow the google one for java - see the wiki) like lack of space between closing and opening brancets: ){
- cohortNumber can probably be just int